src/profiler/profiler.c

```diff
@@ -444,7 +444,7 @@ void xdebug_profiler_function_end(function_stack_entry *fse)
 		fse->profile.time -= call_entry->time_taken;
 		fse->profile.memory -= call_entry->mem_used;
 	}
-	fprintf(XG_PROF(profile_file), "%d %lu %ld\n", fse->profiler.lineno, (unsigned long) (fse->profile.time * 1000000), (fse->profile.memory));
+	fprintf(XG_PROF(profile_file), "%d %lu %ld\n", fse->profiler.lineno, (unsigned long) (fse->profile.time * 1.0e9), (fse->profile.memory));
```

Comment:
this change itself is not enought, because:
- `gettimeofday` has limited precision (no better than us usually)
- epoch time stored as double has about us precision because of limited double precision